core/controller/src/main/scala/org/apache/openwhisk/core/controller/Actions.scala

```diff
@@ -67,7 +67,7 @@ object WhiskActionsApi {
    * 2. An [[execAnnotation]] consistent with the action kind; this annotation is always added and overrides a pre-existing value
    */
   protected[core] def amendAnnotations(annotations: Parameters, exec: Exec, create: Boolean = true): Parameters = {
-    val newAnnotations = if (create) {
+    val newAnnotations = if (create && FeatureFlags.requireApiKeyAnnotation) {
```

Comment:
i suspect that if this is set to false, some tests will fail.

did you consider also updating this for completeness although there may be another place or two that don't use this method that could be affected.

https://github.com/apache/incubator-openwhisk/blob/fb0bab64de6634725b409a63ffbb90fee295e8c1/tests/src/test/scala/org/apache/openwhisk/core/controller/test/ControllerTestCommon.scala#L88-L95

Reply:
For now I do not think that we should make test case complex with feature flag support. So test should work assuming all new features enabled (unless using some SPI mechanism). For main code we just validate manually that if feature is disabled then we get back the earlier behavior.